server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1437,11 +1445,17 @@ public ClusterState getClusterStateUsingDiff(ClusterMetadataManifest manifest, C
             indexRoutingTables.remove(indexName);
         }
 
-        return clusterStateBuilder.stateUUID(manifest.getStateUUID())
+        ClusterState clusterState = clusterStateBuilder.stateUUID(manifest.getStateUUID())
             .version(manifest.getStateVersion())
             .metadata(metadataBuilder)
             .routingTable(new RoutingTable(manifest.getRoutingTableVersion(), indexRoutingTables))
             .build();
+
+        final long durationMillis = TimeValue.nsecToMSec(relativeTimeNanosSupplier.getAsLong() - startTimeNanos);
+        remoteStateStats.stateDownloadSucceeded();
```

Comment:
Should we have separate stats to track count of full download and diff download?

Reply:
+1

Reply:
Added more stats for download:
```
                    "remote_download": {
                        "success_count": 0,
                        "failed_count": 0,
                        "total_time_in_millis": 0,
                        "full_download": 0,
                        "diff_download": 0
                    }

```

Reply:
Would it be better to have success/fail/time individually for full state and diff states?

Reply:
updated.